src/de/mod.rs

```diff
@@ -103,41 +103,34 @@ impl<'a> Deserializer<'a> {
 
     fn parse_string(&mut self) -> Result<String> {
         let start = self.index;
+        let mut backslash = false;
+        let mut escaped = false;
         loop {
             match self.peek() {
                 Some(b'"') => {
-                    // Counts the number of backslashes in front of the current index.
-                    //
-                    // "some string with \\\" included."
-                    //                  ^^^^^
-                    //                  |||||
-                    //       loop run:  4321|
-                    //                      |
-                    //                   `index`
-                    //
-                    // Since we only get in this code branch if we found a " starting the string and `index` is greater
-                    // than the start position, we know the loop will end no later than this point.
-                    let leading_backslashes = |index: usize| -> usize {
-                        let mut count = 0;
-                        loop {
-                            if self.slice[index - count - 1] == b'\\' {
-                                count += 1;
-                            } else {
-                                return count;
-                            }
-                        }
-                    };
-
-                    let is_escaped = leading_backslashes(self.index) % 2 == 1;
-                    if is_escaped {
+                    if escaped {
+                        escaped = false;
                         self.eat_char(); // just continue
                     } else {
                         let end = self.index;
                         self.eat_char();
-                        return unescape::unescape(&self.slice[start..end]);
+                        return if backslash {
+                            unescape::unescape(&self.slice[start..end])
+                        } else {
+                            String::from_utf8(Vec::from(&self.slice[start..end]))
+                                .map_err(|_| Error::InvalidUnicodeCodePoint)
+                        };
```

Comment:
Not sure how much this helps to be honest, since the resulting structs need owned Strings anyways. Maybe this only helps if you deserialize into structs with reference fields.

Reply:
This is explained here: https://serde.rs/lifetimes.html#transient-borrowed-and-owned-data

Reply:
I see. Will try to implement this.

Reply:
With these changes, improvements (compared against the original serde-json-wasm v0.2.2) are up to ~35%(!)

Reply:
Oh, right. We have Visitors in cosmwasm that do not benefit from getting an owned string and thus do not implement `visit_string`. See `impl<'de> de::Visitor<'de> for Base64Visitor`, `impl<'de> de::Visitor<'de> for DecimalVisitor` and `impl<'de> de::Visitor<'de> for Uint128Visitor`